Atomic.md

```diff
@@ -0,0 +1,278 @@
+         be retriable as a second attempt should see the IsEmpty bit on
+         the parent folder and call WritePlaceholderDirectory again. No
+         incomplete placeholders were left in the parent directory.
+
+-   We crash between mkdir() and InitializeEmptyPlaceholder
+
+    -   Placeholder folder does not have flags for virtualization root and
+        empty.
+
+    -   This folder will never be expanded as the kext will exit when
+         we do not see the InVirtualizationRoot bit
+
+-   One of the SetBitInFileFlags calls fails in
+     InitializeEmptyPlaceholder
+
+    -   Folder is marked as in virtualization root but not empty.
+
+        -   This folder will never be expanded as the kext will exit
+             when we do not see the IsEmpty bit
+
+    -   Folder is marked as empty but not in virtualization root
+
+        -   This folder will never be expanded as the kext will exit
+             when we do not see the InVirtualizationRoot bit (we'll
+             never reach the IsEmpty test)
+
+WritePlaceholderFile:
+
+-   Fopen() fails to create a file
+
+    -   This should be retriable as a second attempt should see the
+         IsEmpty bit on the parent folder and call WritePlaceholderFile
+         again. No incomplete placeholders were left in the parent
+         directory.
+
+-   Ftruncate fails to expand the file to the correct size (or is never
+     reached because of a crash)
+
+    -   We have a incomplete placeholder file in the virtualization root
+         that cannot be recovered. Future attempts of
+         WritePlaceholderFile will fail because fopen(path, "wbx")
+         won't let us overwrite the existing file.
+
+-   Memcpy'ing the providerID xattr fails (or is never reached because
+     of a crash)
+
+    -   We have a incomplete placeholder file in the virtualization root
+        that cannot be recovered. There are no flags set on this file.
+
+    -   The MirrorProvider doesn't currently use the providerID for
+         anything (although it should fail if it's empty)
+
+    -   VFSForGit will be unable to parse the placeholder version from
+         the providerID, OnGetFileStream will fail so hydrating this
+         file will fail. Retrying will not help.
+
+-   Memcpy'ing the contentID xattr fails (or is never reached because of
+     a crash)
+
+    -   We have a incomplete placeholder file in the virtualization root
+        that cannot be recovered. There are no flags set on this file.
+
+
+    -   The MirrorProvider doesn't use contentID so no issues will occur
+
+    -   VFSForGit will be unable to get the SHA of the object that it
+         needs to download and hydrating this file will fail. Retrying
+         will not help.
+
+-   We crash before reaching InitializeEmptyPlaceholder
+
+    -   Placeholder does not have flags for virtualization root and
+        empty.
+
+    -   This file will never be expanded as the kext will exit when we
```

Comment:
And will manifest as a file that is the right size but is missing all of its contents